UserTools/RunControl/RunControl.cpp

```diff
@@ -207,14 +252,33 @@ std::string RunControl::RunStop(const char* key){
 
   if(!m_data->running) return "ERROR: Detector not running";
   if(key!="N"){
+    m_data->sc_vars.AlertSend("RunStop");
```

Comment:
Funnily enough you comment this on a duplicate line, so well spotted. :+1: i'll remove that too.
But the function returns a status, which means it *must* be checked. 
Right now `AlertSend` invokes `zmq::socket_t::send`, which returns a value, which `AlertSend` propagates back.
If `zmq::socket_t::send` can never fail in the context of `AlertSend` (e.g. due to only being called with a `ZMQ_PUB` socket), then the signature in [`SlowControlCollection`](https://github.com/ToolDAQ/ToolDAQFramework/blob/main/src/ServiceDiscovery/SlowControlCollection.cpp#L421) should change.

I do note that whether or not `zmq::socket_t::send` can return false on a `ZMQ_PUB` socket or not, i know it can throw (for example if the socket is not connected, or the context is not valid, among many others...) and we should probably be catching those in `SlowControlCollection::AlertSend`.

Reply:
perhaps we should catch them, and return false. :upside_down_face: